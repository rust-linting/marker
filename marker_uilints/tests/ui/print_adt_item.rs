pub enum PrintMeEnum {
    H = -1,
    E = 0,
    R = 17,
}

fn main() {}
