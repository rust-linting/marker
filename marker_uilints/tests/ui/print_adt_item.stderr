warning: printing item
 --> $DIR/print_adt_item.rs:1:10
  |
1 | pub enum PrintMeEnum {
  |          ^^^^^^^^^^^
  |
  = note: Enum(
              EnumItem {
                  data: CommonItemData {
                      id: ItemId(..),
                      span: SpanId(..),
                      vis: Visibility {{ /* WIP: See rust-marker/marker#26 */}},
                      ident: Ident {
                          name: "PrintMeEnum",
                          span: $DIR/print_adt_item.rs:1:10 - 1:21,
                      },
                  },
                  generics: SynGenericParams {
                      params: [],
                      clauses: [],
                  },
                  variants: [
                      EnumVariant {
                          id: VariantId(..),
                          ident: SymbolId(..),
                          span: SpanId(..),
                          kind: Unit,
                          discriminant: Some(
                              ConstExpr {
                                  expr: UnaryOp(
                                      UnaryOpExpr {
                                          data: CommonExprData {
                                              _lifetime: PhantomData<&()>,
                                              id: ExprId(..),
                                              span: SpanId(..),
                                          },
                                          expr: IntLit(
                                              IntLitExpr {
                                                  data: CommonExprData {
                                                      _lifetime: PhantomData<&()>,
                                                      id: ExprId(..),
                                                      span: SpanId(..),
                                                  },
                                                  value: 1,
                                                  suffix: None,
                                              },
                                          ),
                                          kind: Neg,
                                      },
                                  ),
                              },
                          ),
                      },
                      EnumVariant {
                          id: VariantId(..),
                          ident: SymbolId(..),
                          span: SpanId(..),
                          kind: Unit,
                          discriminant: Some(
                              ConstExpr {
                                  expr: IntLit(
                                      IntLitExpr {
                                          data: CommonExprData {
                                              _lifetime: PhantomData<&()>,
                                              id: ExprId(..),
                                              span: SpanId(..),
                                          },
                                          value: 0,
                                          suffix: None,
                                      },
                                  ),
                              },
                          ),
                      },
                      EnumVariant {
                          id: VariantId(..),
                          ident: SymbolId(..),
                          span: SpanId(..),
                          kind: Unit,
                          discriminant: Some(
                              ConstExpr {
                                  expr: IntLit(
                                      IntLitExpr {
                                          data: CommonExprData {
                                              _lifetime: PhantomData<&()>,
                                              id: ExprId(..),
                                              span: SpanId(..),
                                          },
                                          value: 17,
                                          suffix: None,
                                      },
                                  ),
                              },
                          ),
                      },
                  ],
              },
          )
  = note: `#[warn(marker::test_lint)]` on by default

warning: 1 warning emitted

